File: Java/Euler (Java)/src/base/Single.java
package base;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

import problems.P051xP075.P070;
import util.Stopwatch;

public class Single {
	static Problem p = new P070();
	static int runs = 5;
	
	public static void main(String[] args) 
	{
		Config.INSTANCE.init();
		
		Stopwatch s = new Stopwatch();
		s.start("total");
		
		for (int v = 0; v < p.getVariations(); v++)
		{
			if (Config.INSTANCE.loglvl() > 1)
				System.out.println("   Variation " + v + "/" + p.getVariations());
			List<Long> runTimes = new ArrayList<Long>();
			List<Long> execTimes = new ArrayList<Long>();
			
			for (int ii = 0; ii < runs; ii++)
			{
				if (Config.INSTANCE.loglvl() > 1)
					System.out.println("      Iteration " + ii + "/" + runs);
				s.start("iter");
				p.run(v);
				runTimes.add(s.stop("iter"));
				execTimes.add(p.getExecuteTimeNs());
			}
			
			Collections.sort(runTimes);
			Collections.sort(execTimes);
			
			long runNs = runTimes.get((runs-1)/2);
			long execNs = execTimes.get((runs-1)/2);
			
			if (Config.INSTANCE.loglvl() > 0) // human readable
			{
				System.out.print("Problem: " + p.getProblemNumber());
				if (v >= 0)
					System.out.print("_" + v);
				System.out.print(", Run Time(ms): " + (runNs/1000000.0));
				System.out.println(", Exec Time(ms): " + (execNs/1000000.0));
			} else {
				System.out.print(p.getProblemNumber());
				if (v >= 0)
					System.out.print("_" + v);
				System.out.print("," + (runNs/1000000.0));
				System.out.println("," + (execNs/1000000.0));
			}
		}
		
		System.out.println();
		System.out.println("-- Total Time: " + Stopwatch.ppNanos(s.stop("total")));
	}
}

File: C++/Euler (C++)/src/single.cpp
//============================================================================
// Name        : Euler.cpp
// Author      : Andrew McLauthlin
// Version     : 0.1
// Description : Single problem run in Project Euler
//============================================================================

#include <stdlib.h>
#include <algorithm>  // sort
#include <iostream>

#include "base/config.h"
#include "base/timer.h"
//#include "problems/P001xP010/P009.cpp"
//#include "problems/P011xP020/P020.cpp"
//#include "problems/P021xP030/P029.cpp"
#include "problems/P031xP040/P037.cpp"

int main(void)
{
   Config::Instance(); // initialize config
   Problem *p = new P037(); // initialize the problem space

   // Set up the timer
   Timer *t = new Timer();
   t->start();

   Timer *t2 = new Timer();

   for (int v = 0; v < p->getVariations(); v++)
   {
      long runTimes[5];
      long execTimes[5];

      if (Config::LOGLVL() > 1)
         std::cout << "   Variation " << v << "/" << p->getVariations() << std::endl;

      for (int ii = 0; ii < 5; ii++)
      {
         t2->start();
         p->run(v);
         t2->stop();
         runTimes[ii] = t2->getDurationNs();
         execTimes[ii] = p->getExecutionNs();
      }

      std::sort(&runTimes[0], &runTimes[5]);
      std::sort(&execTimes[0], &execTimes[5]);
      long runNs = runTimes[2];
      long execNs = execTimes[2];

      std::cout.precision(15);
      if (Config::LOGLVL() > 0) // human readable
      {
         std::cout << "Problem: " << p->getProblemNumber();
         if (v >= 0)
            std::cout << "_" << v;
         std::cout << ", Run Time (ms): " << (runNs / 1000000.0);
         std::cout << ", Exec Time (ms): " << (execNs / 1000000.0) << std::endl;
      }
      else
      {
         std::cout << p->getProblemNumber();
         if (v >= 0)
            std::cout << "_" << p->getProblemNumber();
         std::cout << "," << (runNs / 1000000.0);
         std::cout << "," << (execNs / 1000000.0) << std::endl;
      }
   }

   // Get the execution time
   t->stop();

   std::cout << std::endl;
   std::cout.precision(15);
   std::cout << "-- Total Execution Time(ms): " << (t->getDurationNs() / 1000000.0) << std::endl;
   delete p; // clean up
   delete t;
   delete t2;
   return EXIT_SUCCESS;
}
